configure.ac

```diff
@@ -75,8 +74,16 @@ AS_IF([test "x${enable_arch_native}" != "xno"],[
   SQUID_CC_CHECK_ARGUMENT([squid_cv_check_marchnative],[-march=native])
 ])
 
-# might be cross-compiling.
-# NP: BUILDCXXFLAGS defined at the end of configure after CXXFLAGS fully known.
+# If the user did not specify a C++ version.
+user_cxx=`echo "$PRESET_CXXFLAGS" | grep -o -E "(-)std="`
+AS_IF([test "x$user_cxx" = "x"],[
+  # Check for C++17 compiler support
+  # May change CXX.
+  AX_CXX_COMPILE_STDCXX([17],[noext],[mandatory])
```

Comment:
We used to call AX_CXX_COMPILE_STDCXX_11() rather than the more general AX_CXX_COMPILE_STDCXX() macro that this PR is using. Autoconf-archive merged all individual version-testing macros into AX_CXX_COMPILE_STDCXX(). Modern individual macros simply call the general one. I think it is better to use the general macro because it saves us from embedding an extra m4 source file (each of those macros, even one-liners, have a dedicated m4 file).